src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1527,6 +1527,16 @@ void Session::cancel_resumption_delay()
     if (unbind_process_complete())
         initiate_rebind(); // Throws
 
+    try {
```

Comment:
How is this different from https://github.com/realm/realm-core/pull/7831 ?

Reply:
It probably wasn't the best (or correct) way to do it, but PR #7831 was branched off of this PR branch, since it needed tests from this PR in order to validate the changes...